File: frontend/types/skin.ts
export interface Skin {
  id: string;
  name: string;
  description: string | null;
  imageUrl: string | null;
  unlockType: string;
  unlockValue: number | null;
  relatedStoneType: string | null;
  isDefault: boolean;
  createdAt: string;
}

export interface UserSkin {
  id: string;
  userId: string;
  skinId: string;
  unlockedAt: string;
}

File: backend/app/models/skin.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Skin(Base):
    __tablename__ = "skins"

    id: Mapped[str] = mapped_column(String, primary_key=True)
    name: Mapped[str] = mapped_column(String, nullable=False)
    description: Mapped[str] = mapped_column(String, nullable=False)
    unlock_type: Mapped[str] = mapped_column(String, nullable=False)
    is_default: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
